Bottom: daf0bf9a68aacfb0c0608e4768cc407d06d8f336
Top:    0ba9cb600d8bc0c356ae769a515c044b0e0a9c96
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-01-19 10:55:30 +0800

Refresh of tun-silent-sparse-warnings

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index a87bfe1..40204f8 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -2230,7 +2230,7 @@ static int __tun_set_ebpf(struct tun_struct *tun, struct tun_prog **prog_p,
 	}
 
 	spin_lock_bh(&tun->lock);
-	old = rcu_dereference_protected(*prog_p,
+	old = rcu_dereference_protected((struct tun_prog __rcu *)*prog_p,
 					lockdep_is_held(&tun->lock));
 	rcu_assign_pointer(*prog_p, new);
 	spin_unlock_bh(&tun->lock);
